applications/ULFapplication/python_scripts/runge_kutta_frac_step_solver.py

```diff
@@ -0,0 +1,225 @@
+            self.elem_neighbor_finder = FindElementalNeighboursProcess(model_part, 2, 10)
+
+
+        self.mark_fluid_process = MarkFluidProcess(model_part);
+
+        self.UlfUtils = UlfUtils()
+
+    #
+    def Initialize(self):
+        (self.neighbour_search).Execute()
+        # calculate the normals to the overall domain
+        self.normal_tools.CalculateOnSimplex(
+            self.model_part.Conditions,
+            self.domain_size)
+        # for SLIP condition we need to save these Conditions in a list
+        # by now SLIP conditions are identified by FLAG_VARIABLE=3.0. this is
+        # done in the constructir of the strategy
+
+        # creating the solution strategy
+        if (self.domain_size == 2):
+            self.solver = RungeKuttaFracStepStrategy2D(
+                self.model_part, self.linear_solver, self.CalculateReactionFlag,
+                self.ReformDofSetAtEachStep, self.CalculateNormDxFlag)
+        if (self.domain_size == 3):
+            self.solver = RungeKuttaFracStepStrategy3D(
+                self.model_part, self.linear_solver, self.CalculateReactionFlag,
+                self.ReformDofSetAtEachStep, self.CalculateNormDxFlag)
+
+        (self.solver).SetEchoLevel(self.echo_level)
+
+        (self.neigh_finder).Execute()
+
+        (self.fluid_neigh_finder).Execute();
+
+        (self.ulf_apply_bc_process).Execute();
+
+        for node in self.model_part.Nodes:
+            node.SetSolutionStepValue(IS_FREE_SURFACE,0,0.0)
+
+        self.RemeshAux();
+
+    def Solve(self):
+
+        (self.solver).SolveStep1()
+
+        self.RemeshAux();
+
+        (self.neigh_finder).Execute()
+        
+        for node in (self.model_part).Nodes:
+            node.Free(PRESSURE)
+            node.Free(VELOCITY_X)
+            node.Free(VELOCITY_Y)
+            node.Free(VELOCITY_Z)
+            if(node.GetSolutionStepValue(IS_FREE_SURFACE)== 1.0):
+                node.SetSolutionStepValue(PRESSURE,0,0.0)
+                node.Fix(PRESSURE)
+            if node.GetSolutionStepValue(IS_STRUCTURE)==1.0:
+                node.Fix(VELOCITY_X)
+                node.Fix(VELOCITY_Y)
+                node.Fix(VELOCITY_Z)
+                node.SetSolutionStepValue(VELOCITY_X,0,0.0)
+                node.SetSolutionStepValue(VELOCITY_Y,0,0.0)
+                node.SetSolutionStepValue(VELOCITY_Z,0,0.0)
+
+        (self.solver).SolveStep2()
+
+        (self.solver).SolveStep3()
+
+    #
+    def SetEchoLevel(self, level):
+        (self.solver).SetEchoLevel(level)
+
+    def RemeshAux(self):
+	
+
+        alpha_shape=1.4;
+
+        h_factor=0.2
+
+        if(self.domain_size == 2):
+            for node in (self.model_part).Nodes:
+                node.SetSolutionStepValue(NODAL_H,0,0.002)
+
+
+        self.node_erase_process = NodeEraseProcess(self.model_part);
+        box_corner1 = Vector(3);
+        box_corner2 = Vector(3);
+
+
+        if(self.domain_size == 2):
+            box_corner1[0]=-0.0001
```

Comment:
many hardcoded values, how do I know what it is the meaning of these?, please use Parameters

Reply:
Thanks. We will change it. @juliobarna  could you please take care of this one?